packages/blade/src/components/Link/BaseLink/BaseLink.tsx

```diff
@@ -0,0 +1,271 @@
+import type { DotNotationSpacingStringToken } from '../../../_helpers/types';
+import type { Theme } from '../../BladeProvider';
+import { useTheme } from '../../BladeProvider';
+import Box from '../../Box';
+import type { IconComponent, IconProps } from '../../Icons';
+import type { BaseTextProps } from '../../Typography/BaseText';
+import BaseText from '../../Typography/BaseText';
+import StyledBaseLink from './StyledBaseLink';
+import useInteraction from './useInteraction';
+
+type BaseLinkCommonProps = {
+  intent?: 'positive' | 'negative' | 'notice' | 'information' | 'neutral';
+  contrast?: 'low' | 'high';
+  icon?: IconComponent;
+  iconPosition?: 'left' | 'right';
+  onClick?: (event: SyntheticEvent) => void;
+  accessibilityLabel?: string;
+};
+
+/*
+  Mandatory children prop when icon is not provided
+*/
+type BaseLinkWithoutIconProps = BaseLinkCommonProps & {
+  icon?: undefined;
+  children: string;
+};
+
+/*
+  Optional children prop when icon is provided
+*/
+type BaseLinkWithIconProps = BaseLinkCommonProps & {
+  icon: IconComponent;
+  children?: string;
+};
+
+/*
+  BaseLink Props with or without an icon
+*/
+type BaseLinkPropsWithOrWithoutIcon = BaseLinkWithIconProps | BaseLinkWithoutIconProps;
+
+/*
+  BaseLink Props when variant is anchor
+*/
+type BaseLinkAnchorVariantProps = BaseLinkPropsWithOrWithoutIcon & {
+  variant?: 'anchor';
+  href?: string;
+  target?: string;
+  rel?: string;
+  isDisabled?: undefined;
+};
+
+/*
+  BaseLink Props when variant is button
+*/
+type BaseLinkButtonVariantProps = BaseLinkPropsWithOrWithoutIcon & {
+  variant?: 'button';
+  isDisabled?: boolean;
+  href?: undefined;
+  target?: undefined;
+  rel?: undefined;
+};
+
+/*
+  BaseLink Props when variant is anchor or button
+*/
+export type BaseLinkProps = BaseLinkAnchorVariantProps | BaseLinkButtonVariantProps;
+
+type BaseLinkStyleProps = {
+  as: 'a' | 'button';
+  textDecorationLine: 'underline' | 'none';
+  iconColor: IconProps['color'];
+  iconPadding: DotNotationSpacingStringToken;
+  textColor: BaseTextProps['color'];
+  focusRingColor: string;
+  motionDuration: DurationString;
+  motionEasing: EasingString;
+  cursor: CSSObject['cursor'];
+  disabled: boolean;
+  role: 'button' | 'link';
+};
+
+const getColorToken = ({
+  variant,
+  intent,
+  contrast,
+  element,
+  currentInteraction,
+  isDisabled,
+  isVisited,
+}: {
+  variant: BaseLinkProps['variant'];
+  intent: BaseLinkProps['intent'];
+  contrast: BaseLinkProps['contrast'];
+  element: 'icon' | 'text';
+  currentInteraction: keyof LinkActionStates;
+  isDisabled: boolean;
+  isVisited: boolean;
+}): IconProps['color'] | BaseTextProps['color'] => {
+  let state = currentInteraction;
+  if (isDisabled && variant == 'button') {
+    state = 'disabled';
+  }
+  if (isVisited && variant == 'anchor' && !intent) {
+    // visited state is only valid for anchor variant without any intent
+    state = 'visited';
+  }
+
+  if (intent && contrast && state !== 'visited') {
```

Comment:
what happens if we have intent but no contrast defined? we shall fall back to `lowContrast` right? in that case we can remove this check from here.

Reply:
Yeah, it will be lowContrast by default but cant remove the whole check, we can just remove `contrast` check and make it

```jsx
  if (intent && state !== 'visited') {
```